### PR TITLE
Omega Cartridges Uplink rounds for CHIMP.

### PR DESCRIPTION
**Pull request by brainfood1183**
Adds scientist specific Uplink item - box of 10 omega cartridges. These are used in the CHIMP and deal 20 burn and will cause anomalies to increase in severity, their cost is 4tc.

untitled-215

https://user-images.githubusercontent.com/113240905/234133855-d492c05c-4d20-48df-a6e9-d52bea0f86d9.mp4

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Treacherous scientists may now spend their TC on Omega Cartridges for the CHIMP. Go APE.

### CODE REVIEW

File: Resources/Locale/en-US/anomaly/anomaly.ftl [hunk 1]
```diff
@@ -8,6 +8,7 @@ anomaly-vessel-component-upgrade-output = point output
 anomaly-particles-delta = Delta particles
 anomaly-particles-epsilon = Epsilon particles
 anomaly-particles-zeta = Zeta particles
+anomaly-particles-omega = Omega particles
```
Comment (EmoGarbage404) [inline Resources/Locale/en-US/anomaly/anomaly.ftl]: unused

File: Content.Server/Anomaly/Components/AnomalousParticleComponent.cs [hunk 2]
```diff
@@ -49,4 +49,13 @@ public sealed class AnomalousParticleComponent : Component
     /// </summary>
     [DataField("stabilityPerWeakeningeHit")]
     public float StabilityPerWeakeningeHit = -0.1f;
+
+    [DataField("AnomalyStabilizer")]
+    public bool AnomalyStabilizer = false;
```
Comment (EmoGarbage404) [inline Content.Server/Anomaly/Components/AnomalousParticleComponent.cs:54]: this should be `AnomalyDestabilizer`, considering that's what it does,

also, the naming for these three variables could be more concise. Perhaps:
- DestabilzingOverride
- WeakeningOverride
- SeverityOverride

Comment (EmoGarbage404) [reply Content.Server/Anomaly/Components/AnomalousParticleComponent.cs:54]: also these all should be documented. Just explain that each one overrides the behavior for having the specific particle type.

File: Content.Server/Anomaly/AnomalySystem.cs [hunk 3]
```diff
@@ -75,15 +75,15 @@ private void OnStartCollide(EntityUid uid, AnomalyComponent component, ref Start
         // small function to randomize because it's easier to read like this
         float VaryValue(float v) => v * Random.NextFloat(MinParticleVariation, MaxParticleVariation);
 
-        if (particle.ParticleType == component.DestabilizingParticleType)
+        if (particle.ParticleType == component.DestabilizingParticleType || particle.AnomalyStabilizer)
         {
             ChangeAnomalyStability(uid, VaryValue(particle.StabilityPerDestabilizingHit), component);
         }
-        else if (particle.ParticleType == component.SeverityParticleType)
+        else if (particle.ParticleType == component.SeverityParticleType || particle.AnomalySeverityIncreaser)
```
Comment (EmoGarbage404) [inline Content.Server/Anomaly/AnomalySystem.cs:82]:
```suggestion
        if (particle.ParticleType == component.SeverityParticleType || particle.AnomalySeverityIncreaser)
```

File: Content.Server/Anomaly/AnomalySystem.cs [hunk 4]
```diff
@@ -75,15 +75,15 @@ private void OnStartCollide(EntityUid uid, AnomalyComponent component, ref Start
...
         {
             ChangeAnomalySeverity(uid, VaryValue(particle.SeverityPerSeverityHit), component);
         }
-        else if (particle.ParticleType == component.WeakeningParticleType)
+        else if (particle.ParticleType == component.WeakeningParticleType || particle.AnomalyWeakener)
```
Comment (EmoGarbage404) [inline Content.Server/Anomaly/AnomalySystem.cs:86]:
```suggestion
        if (particle.ParticleType == component.WeakeningParticleType || particle.AnomalyWeakener)
```

File: Resources/Locale/en-US/store/uplink-catalog.ftl [hunk 5]
```diff
@@ -163,6 +163,9 @@ uplink-banana-peel-explosive-desc = They will burst into laughter when they slip
 uplink-hot-potato-name = Hot Potato
 uplink-hot-potato-desc = Once activated, this time bomb can't be dropped - only passed to someone else!
 
+uplink-chimp-ammo-name = Box of 10 omega cartridges.
+uplink-chimp-ammo-desc = A box of 10 omega particle cartridges for the CHIMP, omega particles cause severe burns and anomalies to go critical.
```
Comment (EmoGarbage404) [inline Resources/Locale/en-US/store/uplink-catalog.ftl]:
```suggestion
uplink-chimp-ammo-desc = A box of 10 omega particle cartridges for the CHIMP. Omega particles inflict severe burns and cause anomalies to go supercritical.
```

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml [hunk 6]
```diff
@@ -397,6 +397,31 @@
...
+  components:
+  - type: Sprite
+    sprite: Objects/Weapons/Guns/Projectiles/magic.rsi
+    layers:
+      - state: magicm_green
+        shader: unshaded
+  - type: AnomalousParticle
+    particleType: Default
+    anomalyStabilizer: true
+    anomalyWeakener: true
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:413]: if you include both of these, they'll balance out. Just remove the weakening effect.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]
nit and picks

File: Resources/Locale/en-US/anomaly/anomaly.ftl (hunk 1 above)
Comment (brainfood1183) [reply Resources/Locale/en-US/anomaly/anomaly.ftl]: removed it.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml (hunk 6 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:413]: will do

File: Content.Server/Anomaly/Components/AnomalousParticleComponent.cs (hunk 2 above)
Comment (brainfood1183) [reply Content.Server/Anomaly/Components/AnomalousParticleComponent.cs]: added little description for each.

File: Resources/Locale/en-US/store/uplink-catalog.ftl (hunk 5 above)
Comment (brainfood1183) [reply Resources/Locale/en-US/store/uplink-catalog.ftl]: made this change

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml (hunk 6 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml]: done

File: Content.Server/Anomaly/Components/AnomalousParticleComponent.cs [hunk 7]
```diff
@@ -49,4 +49,22 @@ public sealed class AnomalousParticleComponent : Component
...
+
+    /// <summary>
+    /// If this is true then the particle will always effect the weakeness of the anomaly.
+    /// </summary>
+    [DataField("weakeningOverride")]
+    public bool WeakeningOverride = false;
+
+    /// <summary>
+    /// If this is true then the particle will always effect the severity of the anomaly.
+    /// </summary
```
Comment (EmoGarbage404) [inline Content.Server/Anomaly/Components/AnomalousParticleComponent.cs:67]:
```suggestion
    /// </summary>
```

**Review by EmoGarbage404 — APPROVED** [review]

**Comment by brainfood1183** [comment]
> nit and picks



> nit and picks

should be all sorted now, might just want to check make sure no unwanted code, i accidentally messed up my pr and had to sort it all. Tested and working though.

File: Content.Server/Anomaly/Components/AnomalousParticleComponent.cs [hunk 8]
```diff
@@ -49,4 +49,22 @@ public sealed class AnomalousParticleComponent : Component
     /// </summary>
     [DataField("stabilityPerWeakeningeHit")]
     public float StabilityPerWeakeningeHit = -0.1f;
+
+    /// <summary>
+    /// If this is true then the particle will always effect the stability of the anomaly.
+    /// </summary>
+    [DataField("destabilzingOverride")]
+    public bool DestabilzingOverride = false;
```
Comment (metalgearsloth) [inline Content.Server/Anomaly/Components/AnomalousParticleComponent.cs:57]: Destabilizing and affect.

File: Content.Server/Anomaly/Components/AnomalousParticleComponent.cs [hunk 9]
```diff
@@ -49,4 +49,22 @@ public sealed class AnomalousParticleComponent : Component
...
+
+    /// <summary>
+    /// If this is true then the particle will always effect the weakeness of the anomaly.
+    /// </summary>
```
Comment (metalgearsloth) [inline Content.Server/Anomaly/Components/AnomalousParticleComponent.cs:61]: weakness, also affect.

File: Content.Server/Anomaly/Components/AnomalousParticleComponent.cs [hunk 10]
```diff
@@ -49,4 +49,22 @@ public sealed class AnomalousParticleComponent : Component
...
+    [DataField("weakeningOverride")]
+    public bool WeakeningOverride = false;
+
+    /// <summary>
+    /// If this is true then the particle will always effect the severity of the anomaly.
+    /// </summary>
```
Comment (metalgearsloth) [inline Content.Server/Anomaly/Components/AnomalousParticleComponent.cs:67]:
```suggestion
    /// <summary>
    /// If this is true then the particle will always affect the severity of the anomaly.
    /// </summary>
```

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**